integrations/lib/testing/integration/authhelper.go

```diff
@@ -0,0 +1,197 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package integration
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/constants"
+	"github.com/gravitational/teleport/api/identityfile"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/keys"
+	libauth "github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+	"testing"
+	"time"
+)
+
+// OSSAuthHelper implements the AuthHelper interface.
+// It starts an OSS Auth server and exposes the functions required for plugin
+// integration tests to build teleport clients for each user/plugin/bot.
+type OSSAuthHelper struct {
+	server *libauth.TestTLSServer
+	dir    string
+}
+
+// StartServer implements the AuthHelper interface.
+func (a *OSSAuthHelper) StartServer(t *testing.T) *client.Client {
+	a.dir = t.TempDir()
+	authServer, err := libauth.NewTestAuthServer(libauth.TestAuthServerConfig{
+		Dir: a.dir,
+	})
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		require.NoError(t, authServer.Close())
+	})
+
+	server, err := libauth.NewTestTLSServer(libauth.TestTLSServerConfig{
+		APIConfig: &libauth.APIConfig{
+			AuthServer: authServer.AuthServer,
+			Authorizer: authServer.Authorizer,
+			AuditLog:   authServer.AuditLog,
+			Emitter:    authServer.AuditLog,
+		},
+		AuthServer:    authServer,
+		AcceptedUsage: authServer.AcceptedUsage,
+	})
+	require.NoError(t, err)
+	a.server = server
+
+	t.Cleanup(func() {
+		require.NoError(t, server.Close())
+	})
+
+	authClient, err := server.NewClient(libauth.TestAdmin())
+	require.NoError(t, err)
+
+	return authClient.APIClient
+}
+
+// ServerAddr implements the AuthHelper interface.
+// It returns the server address, including the port.
+func (a *OSSAuthHelper) ServerAddr() string {
+	return a.server.Addr().String()
+}
+
+// getUserCerts generates a private key for the user and has the auth sign a TLS and an SSH certificate.
+// The function returns three values:
+// - the PEM-encoded private key
+// - the Authorized-key formatted SSH cert
+// - the PEM-encoded TLS cert
+func (a *OSSAuthHelper) getUserCerts(t *testing.T, user types.User) ([]byte, []byte, []byte) {
```

Comment:
Can we have a new type that includes all three fields?
Returning 3 `[]byte` might create some confusion during development